### PR TITLE
Initial Analysis wizard

### PR DESCRIPTION
**Pull request by gildub**
Adds 
- step 1: Analysis mode, transformation paths (targets) and scope
- step 3: review and task creation

Note: Step 2 Advanced (custom rules and options) are optional and coming with follow-up PR.

### CODE REVIEW

File: pkg/client/src/app/api/models.tsx [hunk 1]
```diff
@@ -307,3 +289,73 @@ export interface Setting {
...
+  | "Completed"
+  | "Failed"
+  | "InProgress"
+  | "NotStarted"
+  | "Scheduled";
+
+export interface Task {
+  id?: number;
+  name: string;
+  // updateUser: string;
```
Comment (ibolton336) [inline pkg/client/src/app/api/models.tsx:306]: do we need these?

File: pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      ],
+    },
+    {
+      name: "Advanced",
+      steps: [
+        {
+          name: "Custom rules",
+          component: (
+            <CustomRules
+            // projectId={1}
```
Comment (ibolton336) [inline pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx:149]: comments

File: pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      ],
+    },
+    {
+      name: "Review",
+      component: <Review applications={applications} getValues={getValues} />,
+      nextButtonText: "Run",
+    },
+  ];
+
+  console.log(watch());
```
Comment (ibolton336) [inline pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx:164]: console log

File: pkg/client/src/app/pages/applications/analysis-wizard/components/select-card.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.select-card__component__empty-state {
+  padding: 0 !important;
```
Comment (ibolton336) [inline pkg/client/src/app/pages/applications/analysis-wizard/components/select-card.css:2]: Any reason we aren't using the spacing library here from pf?

File: pkg/client/package.json [hunk 5]
```diff
@@ -38,6 +39,7 @@
     "keycloak-js": "12.0.2",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-hook-form": "^7.27.1",
```
Comment (ibolton336) [inline pkg/client/package.json:42]: Why add this when we have formik?

Comment (gildub) [reply pkg/client/package.json:42]: React-hook-form is much more efficient both in term of library size and rendering times.
So ultimately all formik code to be refactored using it.

File: pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx (hunk 2 above)
Comment (gildub) [reply pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx:149]: Thanks

File: pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx (hunk 3 above)
Comment (gildub) [reply pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx:164]: Removing it

File: pkg/client/src/app/pages/applications/analysis-wizard/components/select-card.css (hunk 4 above)
Comment (gildub) [reply pkg/client/src/app/pages/applications/analysis-wizard/components/select-card.css:2]: That's legacy code from MTA. 
We can replace that.

File: pkg/client/src/app/api/models.tsx (hunk 1 above)
Comment (gildub) [reply pkg/client/src/app/api/models.tsx:306]: Removing it.

**Review by ibolton336 — APPROVED** [review]

**Pull request merged**